packages/Search/src/details/DTK_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -197,99 +199,197 @@ DistributedSearchTreeImpl<DeviceType>::sendAcrossNetwork(
 template <typename DeviceType>
 template <typename Query>
 void DistributedSearchTreeImpl<DeviceType>::deviseStrategy(
-    Point eps, Kokkos::View<Query *, DeviceType> queries,
-    BVH<DeviceType> const &bvh, Kokkos::View<int *, DeviceType> &indices,
-    Kokkos::View<int *, DeviceType> &offset )
+    Kokkos::View<Query *, DeviceType> queries,
+    DistributedSearchTree<DeviceType> const &tree,
+    Kokkos::View<int *, DeviceType> &indices,
+    Kokkos::View<int *, DeviceType> &offset,
+    Kokkos::View<double *, DeviceType> & )
 {
-    int const n_queries = queries.extent( 0 );
-    Kokkos::View<Details::Overlap *, DeviceType> overlap_queries(
-        "overlap_queries", n_queries );
+    auto const &top_tree = tree._top_tree;
+    auto const &bottom_tree_sizes = tree._bottom_tree_sizes;
+
+    // Find the k nearest local trees.
```

Comment:
You should grep for `local` and `distributed` and replace them by `bottom` and `top`

Reply:
Comment is accurate.  I had already renamed all variables.

Reply:
I don't understand. The comments here and below are about local trees but you have renamed them bottom trees, e.g., `Stop if local trees get empty` but the code is `if ( bottom_tree_size == 0 ) `